### PR TITLE
Feature/add map name to custom maps

### PR DESCRIPTION
**Pull request by alexp8**
Custom maps downloaded from CnCNet will contain the name of the map in the filename to allow easier navigation of custom maps.

Approach:
Leave map upload as is, uploading the file with sha1 sum as the filename.
After map is downloaded, rename the map with the map name in the filename.

### CODE REVIEW

**Comment by devo1929** [comment]
two things:

1. I believe someone had raised the concern with the filenames _possibly_ being too long. If this is still a concern, we could always limit the sha to the first 6-10 characters in the filename.
2. There are a lot of locations in the code where the filename is being constructed and we're concatenating "mapName_sha" for each map. It'd be nice if there was a shared "utility" function to construct this name for a passed in mapName and sha, so that it's one location in the code constructing it every time, should it ever need to change again.

**Comment by alexp8** [comment]
> two things:
> 
> 1. I believe someone had raised the concern with the filenames _possibly_ being too long. If this is still a concern, we could always limit the sha to the first 6-10 characters in the filename.
> 2. There are a lot of locations in the code where the filename is being constructed and we're concatenating "mapName_sha" for each map. It'd be nice if there was a shared "utility" function to construct this name for a passed in mapName and sha, so that it's one location in the code constructing it every time, should it ever need to change again.

1. We would want to display the full sha1 sum in the file name to guarantee uniqueness, I'm not aware of any issue where the filenaame could be too long.
2. I can create utility function.

**Comment by alexp8** [comment]
I pushed the utility function

File: DXMainClient/DXGUI/Multiplayer/GameLobby/CnCNetGameLobby.cs [hunk 1]
```diff
@@ -1534,8 +1544,8 @@ private void MapSharer_HandleMapDownloadFailed(SHA1EventArgs e)
 
         private void MapSharer_HandleMapDownloadComplete(SHA1EventArgs e)
         {
-            Logger.Log("Map " + e.SHA1 + " downloaded, parsing.");
-            string mapPath = "Maps/Custom/" + e.SHA1;
+            Logger.Log("Map " + e.MapName + "_" + e.SHA1 + " downloaded, parsing.");
+            string mapPath = "Maps/Custom/" + e.MapName + "_" + e.SHA1;
```
Comment (devo1929) [inline DXMainClient/DXGUI/Multiplayer/GameLobby/CnCNetGameLobby.cs:1548]: two here for use of your new utility function

Comment (alexp8) [reply DXMainClient/DXGUI/Multiplayer/GameLobby/CnCNetGameLobby.cs]: done

File: DXMainClient/DXGUI/Multiplayer/GameLobby/CnCNetGameLobby.cs [hunk 2]
```diff
@@ -973,11 +980,15 @@ private void ApplyGameOptions(string sender, string message)
                 AddNotice("The game host has changed ProtocolVersion to " + protocolVersion);
             }
 
+            string mapName = parts[partIndex + 8];
+
+
             GameMode currentGameMode = GameMode;
             Map currentMap = Map;
```
Comment (Metadorius) [inline DXMainClient/DXGUI/Multiplayer/GameLobby/CnCNetGameLobby.cs:987]:
```suggestion
            string mapName = parts[partIndex + 8];
            GameMode currentGameMode = GameMode;
            Map currentMap = Map;
```

File: DXMainClient/Domain/Multiplayer/CnCNet/MapSharer.cs [hunk 3]
```diff
@@ -363,20 +365,28 @@ private static void Download(object details)
...
                     Download(array);
                 }
             }
         }
 
-        private static string DownloadMain(string sha1, string myGame, out bool success)
+        public static string GetMapFileName(string sha1, string mapName)
+        {
+            return mapName + "_" + sha1;
+        }
```
Comment (Metadorius) [inline DXMainClient/Domain/Multiplayer/CnCNet/MapSharer.cs:381]:
```suggestion
        public static string GetMapFileName(string sha1, string mapName)
            => mapName + "_" + sha1;
```

**Review by Metadorius — APPROVED** [review]
Looks good. Should use expression bodies where it's applicable though. May also refactor the file to use them if you feel like it and there are other cases where it's not applied yet.

**Pull request merged**